### PR TITLE
feat: change template max_ttl to default_ttl

### LINKED ISSUE #3304: Move template schedule controls into enterprise
As an administrative control, this should be in our enterprise version.

The specific flags are:

- `--max_ttl`
- and `--min_autostart_interval`

### LINKED ISSUE #3811: Enterprise FR: allow admins to control the max time-until-shutdown
@sharkymark and I met with a current customer today, and they expressed the need to control the maximum amount of auto-stop time their developers can set.

their devs are currently setting the auto-shutoff time in v1 to `Never`, which is problematic for controlling cloud costs.

### LINKED ISSUE #4849: Purpose of the auto stop limit?
Hallo,

I think I have misunderstood the purpose if this setting in a template.

New Workspaces have no schedule by default and they run forever even if I have set an auto stop limit in the template. I'm looking for functionality to configure a limit even if the user have configured no schedule. I thought that this setting is the solution, but it only limits the Field for auto stop field in the schedule form, right? Or is it a bug?

The description of the field 'Workspaces created from this template may not remain running longer than 5 hours.' is a little bit misleading.

### PR DESCRIPTION
**Pull request by f0ssel**
What this changes:
- Removal of the "auto-stop max" feature on a template
- Removal of the "min auto-start interval" - were not seeing value or usage from it and it simplifies the feature to remove it. 
- Addition of an "auto-stop default" feature - any workspaces made the from the template will start with an auto-off of X hours. Users are free to set the auto-off back to 0 or any other value once created. 
- Migration of the auto-stop max value to the new default - I feel this was safe enough since we are moving to a more lax state and I didn't want people to lose their data. Let me know if anyone disagrees with the decision or has suggestions to improve. 

image

Closes https://github.com/coder/coder/issues/4849 - basic behavior has changed
Closes https://github.com/coder/coder/issues/3304 - this functionality will be provided with the upcoming quotas work
Closes https://github.com/coder/coder/issues/3811 - No longer relevant to this feature

### CODE REVIEW

**Review by Kira-Pilot — APPROVED** [review]
FE ✅

**Review by bpmct — COMMENTED** [review]
LGTM - I agree with all these decisions

**Review by johnstcn — APPROVED** [review]
👍

File: cli/templatecreate.go [hunk 1]
```diff
@@ -133,7 +132,7 @@ func templateCreate() *cobra.Command {
 	cmd.Flags().StringVarP(&directory, "directory", "d", currentDirectory, "Specify the directory to create from")
 	cmd.Flags().StringVarP(&provisioner, "test.provisioner", "", "terraform", "Customize the provisioner backend")
 	cmd.Flags().StringVarP(&parameterFile, "parameter-file", "", "", "Specify a file path with parameter values.")
-	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 24*time.Hour, "Specify a maximum TTL for workspaces created from this template.")
+	cmd.Flags().DurationVarP(&defaultTTL, "default-ttl", "", 24*time.Hour, "Specify a default TTL for workspaces created from this template.")
 	cmd.Flags().DurationVarP(&minAutostartInterval, "min-autostart-interval", "", time.Hour, "Specify a minimum autostart interval for workspaces created from this template.")
```
Comment (mafredri) [inline cli/templatecreate.go:136]: Should this flag be removed too?

File: coderd/database/migrations/000072_remove_min_autostart.up.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- drop "min_autostart_interval" column from "templates" table
+ALTER TABLE "templates" DROP COLUMN "min_autostart_interval";
+
+-- rename "max_ttl" to "default_ttl" on "templates" table
+ALTER TABLE "templates" RENAME COLUMN "max_ttl" TO "default_ttl";
```
Comment (mafredri) [inline coderd/database/migrations/000072_remove_min_autostart.up.sql:5]: I would absolutely ❤️ if you wanted to add comments on this new column!

```sql
COMMENT ON COLUMN templates.max_ttl IS 'yay things';
```

**Review by mafredri — APPROVED** [review]
Lovely change!

File: cli/templatecreate.go (hunk 1 above)
Comment (f0ssel) [reply cli/templatecreate.go:136]: Yes! good catch

**Pull request merged**